include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,919 @@
+    constexpr constexpr_vector & operator=(constexpr_vector const &) = default; //!< Defaulted.
+    constexpr constexpr_vector & operator=(constexpr_vector &&) = default;      //!< Defaulted.
+    ~constexpr_vector() = default;                                              //!< Defaulted.
+
+    /*!\brief Construct from an (smaller or equally sized) array over the same value type.
+     * \param[in] array The array to copy the data from.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr constexpr_vector(std::array<value_type, N> const & array) :
+        data{array}, sz{N}
+    {}
+
+    //!\cond
+    template <size_t N_other>
+    //!\cond
+        requires (N_other <= N)
+    //!\endcond
+    constexpr constexpr_vector(std::array<value_type, N_other> const & array) :
+        sz{N_other}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+    //!\endcond
+
+    /*!\brief Construct from a (smaller or equally sized) built in array over the same value type.
+     * \param[in] array The array to copy the data from.
+     *
+     * ### Complexity
+     *
+     * Linear in `N`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr constexpr_vector(const value_type (&array)[N]) :
+        sz{N}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+
+    //!\cond
+    template <size_t N_other>
+    //!\cond
+        requires (N_other <= N)
+    //!\endcond
+    constexpr constexpr_vector(const value_type (&array)[N_other]) :
+        sz{N_other}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+    //!\endcond
+
+    /*!\brief Construct from a list of values of value_type.
+     * \tparam other_value_type A parameter pack where each type is equal to value_type.
+     * \param[in] args The values to construct from.
+     */
+    template <typename ...other_value_type>
+    //!\cond
+        requires (std::Same<value_type, other_value_type> && ...)
+    //!\endcond
+    constexpr constexpr_vector(other_value_type... args) :
+        data{args...}, sz{sizeof...(other_value_type)}
+    {}
+
+    /*!\brief Construct with `n` times `value`.
+     * \param[in] n     Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * Linear in `n`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr constexpr_vector(size_type n, value_type value) : constexpr_vector{}
+    {
+        this->insert(cbegin(), n, value);
+        sz = n;
+    }
+
+    /*!\brief Construct from two iterators.
+     * \tparam begin_iterator_type Must satisfy std::ForwardIterator and `value_type` must be constructible from
```

Comment:
```suggestion
     * \tparam begin_iterator_type Must model std::ForwardIterator and `value_type` must be constructible from
```

also in other places